### DISCUSSION TITLE
Webpack Dev Server Websocket connection failure

### DISCUSSION THREAD

**Discussion by buck-rivard** [community/community#11524] [Codespaces]
Our application uses create react app's standard webpack development server for hot module reloading. We are able to get HMR to work running locally, but in codespace the websocket connection request to `wss://{codespaces_forwarded_url}/ws` fails. 

```
WebSocket connection to 'wss://***.githubpreview.dev:3000/ws' failed: WebSocketClient.js:16 
```
<img width="1423" alt="IMG_0783 copy" src="https://user-images.githubusercontent.com/91921078/153959041-cefb159f-d940-426d-ab7a-b0af74b92346.png">

**Comment by bchuecos** [comment]
Hi @buck-rivard, thank you for the feedback.

~~It looks like your react app is trying to reach `wss://{codespaces_forwarded_url}/ws`, but we don't currently support forwarding ports in Codespaces in protocols outside of http/https. We are however actively working on supporting new protocols soon to support use cases like this.~~

Correction: wss is supported today, the issue here is in the URL being used: `'wss://***.githubpreview.dev:3000/ws'`. There's nothing on :3000, this should be port 443, we don't support port access in that way. The web socket should be reachable at `wss://***-3000.githubpreview.dev:443/ws` or `wss://***-3000.githubpreview.dev/ws`. It's confusing because there's two ports here: the port in your Codespace (port 3000) and the port in our service doing the redirect (port 443).

**Reply by markgoho** [reply]
hi @bchuecos could you share a link to a roadmap issue that's tracking this "active work" on supporting new protocols?

**Reply by ehewitt-sel** [reply]
I've been trying to find a work around but as of yet have been unsucessful.
Am I correct in my understanding that every user of React that wants to use GitHub Codespaces will not be able to hot reload because of this limitation?

**Reply by bchuecos** [reply]
Posted correction above

**Reply by ehewitt-sel** [reply]
Thanks, @bchuecos !

For those that might be using webpack, the easiest way to do this is to set the web socket path with an environment variable before running npm start, as follows:

export WDS_SOCKET_PORT=443

**Reply by arthurfanti** [reply]
not sure if I got this right, but CRA doesn't allow webpack customizations, thus we need to use something like craco

**Reply by bartlettpsj** [reply]
This worked for me with Webpacked /  CRA app
`    "start": "export WDS_SOCKET_PORT=443 && react-scripts start",`
